Error when sampling using HMA after changing the default distributions and grandparent - parent - child table relationship

### Environment Details

Please indicate the following details about the environment in which you found the bug:

* SDV version: 1.24
* Python version: 3.12
* Operating System: darwin

### Error Description

When creating an HMA Synthesizer with a dataset that includes a grandparent-parent-child relationship, changing the default distribution of the table synthesizers to 'norm' or 'uniform' causes an error when sampling

### Steps to reproduce
Example
```
data = {
    'child': pd.DataFrame({'id': range(10), 'parent_id': range(10)}),
    'parent': pd.DataFrame({
        'id': range(10),
        'grandparent_id': range(10),
        'categories': list(np.random.choice(['T', 'F'], size=10)),
    }),
    'grandparent': pd.DataFrame({'id': range(10)}),
}
metadata = Metadata.load_from_dict({
    'tables': {
        'child': {
            'primary_key': 'id',
            'columns': {'id': {'sdtype': 'id'}, 'parent_id': {'sdtype': 'id'}},
        },
        'parent': {
            'primary_key': 'id',
            'columns': {
                'id': {'sdtype': 'id'},
                'grandparent_id': {'sdtype': 'id'},
                'categories': {'sdtype': 'categorical'},
            },
        },
        'grandparent': {'primary_key': 'id', 'columns': {'id': {'sdtype': 'id'}}},
    },
    'relationships': [
        {
            'parent_table_name': 'parent',
            'child_table_name': 'child',
            'parent_primary_key': 'id',
            'child_foreign_key': 'parent_id',
        },
        {
            'parent_table_name': 'grandparent',
            'child_table_name': 'parent',
            'parent_primary_key': 'id',
            'child_foreign_key': 'grandparent_id',
        },
    ],
    'METADATA_SPEC_VERSION': 'V1',
})
synthesizer = HMASynthesizer(metadata)

synthesizer.set_table_parameters('parent', {'default_distribution': 'norm'})
synthesizer.fit(data)
synthesizer.sample(1)

```


Traceback
```
sdv/multi_table/base.py:675: in sample
    sampled_data = self._sample(scale=scale)
sdv/sampling/hierarchical_sampler.py:324: in _sample
    self._sample_children(table_name=table, sampled_data=sampled_data, scale=scale)
sdv/sampling/hierarchical_sampler.py:211: in _sample_children
    self._add_child_rows(
sdv/sampling/hierarchical_sampler.py:101: in _add_child_rows
    sampled_rows = self._sample_rows(child_synthesizer, num_rows)
sdv/sampling/hierarchical_sampler.py:76: in _sample_rows
    return synthesizer._sample_batch(round(num_rows), keep_extra_columns=True)
sdv/single_table/base.py:960: in _sample_batch
    sampled, num_valid = self._sample_rows(
sdv/single_table/base.py:863: in _sample_rows
    raw_sampled = self._sample(num_rows)
sdv/single_table/copulas.py:194: in _sample
    return self._model.sample(num_rows, conditions=conditions)
../Copulas/copulas/utils.py:50: in wrapper
    return function(self, *args, **kwargs)
../Copulas/copulas/multivariate/gaussian.py:299: in sample
    output[column_name] = univariate.percent_point(cdf)
../Copulas/copulas/univariate/base.py:593: in percent_point
    return self.MODEL_CLASS.ppf(U, **self._params)
>       args, loc, scale = self._parse_args(*args, **kwds)
E       TypeError: _parse_args() got an unexpected keyword argument 'a'

../miniconda3/envs/sdv/lib/python3.12/site-packages/scipy/stats/_distn_infrastructure.py:2293: TypeError
```

--- a/sdv/multi_table/hma.py
+++ b/sdv/multi_table/hma.py
@@ -565,6 +565,9 @@ class HMASynthesizer(BaseHierarchicalSampler, BaseMultiTableSynthesizer):
                 )
                 synthesizer = self._synthesizer(table_meta, **self._table_parameters[child_name])
 
+            self._set_extended_columns_distributions(
+                synthesizer, child_name, self._parent_extended_columns[child_name]
+            )
             synthesizer._set_parameters(parameters, default_parameters)
         else:
             synthesizer = self._null_child_synthesizers[f'__{child_name}__{foreign_key}']

--- a/tests/integration/multi_table/test_hma.py
+++ b/tests/integration/multi_table/test_hma.py
@@ -1970,8 +1970,59 @@ def test_hma_0_1_grandparent():
     child_df = synthetic_data['child']
     data_col_max = child_df['data'].max()
     data_col_min = child_df['data'].min()
-    assert child_df[child_df['data'] == data_col_max].shape[0] == 2
-    assert child_df[child_df['data'] == data_col_min].shape[0] == 1
+    assert child_df[child_df['data'] == data_col_max].shape[0] == 3
+    assert child_df[child_df['data'] == data_col_min].shape[0] == 3
+
+
+def test_parent_default_distribution_non_beta():
+    # Setup
+    data = {
+        'child': pd.DataFrame({'id': range(10), 'parent_id': range(10)}),
+        'parent': pd.DataFrame({
+            'id': range(10),
+            'grandparent_id': range(10),
+            'categories': list(np.random.choice(['T', 'F'], size=10)),
+        }),
+        'grandparent': pd.DataFrame({'id': range(10)}),
+    }
+    metadata = Metadata.load_from_dict({
+        'tables': {
+            'child': {
+                'primary_key': 'id',
+                'columns': {'id': {'sdtype': 'id'}, 'parent_id': {'sdtype': 'id'}},
+            },
+            'parent': {
+                'primary_key': 'id',
+                'columns': {
+                    'id': {'sdtype': 'id'},
+                    'grandparent_id': {'sdtype': 'id'},
+                    'categories': {'sdtype': 'categorical'},
+                },
+            },
+            'grandparent': {'primary_key': 'id', 'columns': {'id': {'sdtype': 'id'}}},
+        },
+        'relationships': [
+            {
+                'parent_table_name': 'parent',
+                'child_table_name': 'child',
+                'parent_primary_key': 'id',
+                'child_foreign_key': 'parent_id',
+            },
+            {
+                'parent_table_name': 'grandparent',
+                'child_table_name': 'parent',
+                'parent_primary_key': 'id',
+                'child_foreign_key': 'grandparent_id',
+            },
+        ],
+        'METADATA_SPEC_VERSION': 'V1',
+    })
+    synthesizer = HMASynthesizer(metadata)
+    synthesizer.set_table_parameters('parent', {'default_distribution': 'norm'})
+    synthesizer.fit(data)
+
+    # Run
+    synthesizer.sample(1)
 
 
 parametrization = [

--- a/tests/unit/multi_table/test_hma.py
+++ b/tests/unit/multi_table/test_hma.py
@@ -472,6 +472,7 @@ class TestHMASynthesizer:
         instance._table_parameters = {'users': {'a': 1}}
         instance._table_synthesizers = {'users': table_synthesizer}
         instance._default_parameters = {'users': {'colA': 'default_param', 'colB': 'default_param'}}
+        instance._parent_extended_columns = {'users': []}
 
         # Run
         synthesizer = HMASynthesizer._recreate_child_synthesizer(
@@ -493,6 +494,9 @@ class TestHMASynthesizer:
             {'colA': 'default_param', 'colB': 'default_param'},
         )
         instance._extract_parameters.assert_called_once_with(parent_row, table_name, 'session_id')
+        instance._set_extended_columns_distributions.assert_called_once_with(
+            synthesizer, table_name, []
+        )
 
     def test__get_likelihoods(self):
         """Test that ``_get_likelihoods`` computes the likelihoods.

